### PR TITLE
feat: drop support for php < 8.1

### PR DESCRIPTION
**Pull request by Chris53897**
drop support for symfony < 5.4
add "ext-simplexml" as required
add strong types

As i was working in https://github.com/moneyphp/iso-currencies/pull/14
i made some more changes for a new major version. There are BC breaks, and i am not sure about the changes in the interface.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -13,8 +13,9 @@
         }
     ],
     "require": {
-        "php": ">=5.5",
-        "symfony/yaml": "^3.2 | ^4.0 | ^5.0 | ^6.0"
+        "php": ">=8.1",
```
Comment (frederikbosch) [inline composer.json]: ~8.1.0 || ~8.2.0

File: src/Country.php [hunk 2]
```diff
@@ -1,85 +1,37 @@
...
-
-    /**
-     * @param string $entity
-     * @param string $currency
-     * @param string $alphabeticCode
-     * @param int $numericCode
-     * @param int $minorUnit
-     */
-    public function __construct($entity, $currency, $alphabeticCode, $numericCode, $minorUnit)
+    public function __construct(private readonly string $entity,
```
Comment (frederikbosch) [inline src/Country.php]: Add new line

File: src/Country.php [hunk 3]
```diff
@@ -1,85 +1,37 @@
...
+                                private readonly string $currency,
+                                private readonly string $alphabeticCode,
+                                private readonly int    $numericCode,
+                                private readonly int    $minorUnit)
```
Comment (frederikbosch) [inline src/Country.php]: Add new line, identation

**Review by frederikbosch — COMMENTED** [review]
Great work!

**Comment by frederikbosch** [comment]
Let's get this is in, with the minor changes I suggested.

File: composer.json (hunk 1 above)
Comment (Chris53897) [reply composer.json]: You mean ~8.1.0 || ~8.2.0 || ~8.3.0 ?

File: src/Country.php (hunk 3 above)
Comment (Chris53897) [reply src/Country.php]: i changed identation betwen type and variable name. I am not sure if you that is what you mean.

I am not sure where exact you want me to add the new line

**Comment by frederikbosch** [comment]
The two new lines and indentation are still left. If you correct them I will press the button.

**Comment by Chris53897** [comment]
I am not sure how exactly you want it.
I did change it now, but maybe wrong

**Pull request merged**